### ISSUE TITLE
Bug: compiles to an invalid statement when inserting with multiple values involving optional columns

### ISSUE THREAD

**Issue by nktpro** [koskimas/kysely#231]
Thank you for this phenomenal library!

I've run into a bug with query compilation. When a table has multiple optional columns, inserting multiple values, each supplying different columns will result in an invalid query.

Here's a reproducer:

```typescript
import {
  DummyDriver,
  Kysely,
  PostgresAdapter,
  PostgresIntrospector,
  PostgresQueryCompiler,
} from "kysely";

interface TestTable {
  first: string | null;
  second: string | null;
  third: string | null;
}

interface Database {
  test: TestTable;
}

const db = new Kysely<Database>({
  dialect: {
    createAdapter() {
      return new PostgresAdapter();
    },
    createDriver() {
      return new DummyDriver();
    },
    createIntrospector(db: Kysely<unknown>) {
      return new PostgresIntrospector(db);
    },
    createQueryCompiler() {
      return new PostgresQueryCompiler();
    },
  },
});

const query = db
  .insertInto("test")
  .values([
    {
      first: "first",
    },
    {
      second: "second",
    },
    {
      third: "third",
    },
  ])
  .compile();

console.log(query.sql);
```

which will output

```
insert into "test" ("first", "second", "third") values ($1, defaultdefault), (default$2, default), (default, default, $3)
```

Notice that there's a missing comma separator in the first and second group (`defaultdefault` instead of `default, default`, and `default$2` instead of `default, $2`)

**Comment by koskimas** [comment]
This has been fixed, but not released yet.

**Comment by koskimas** [comment]
Fixed in 0.23.0
